apps/dashboard/app/models/motd_formatter_markdown_erb.rb

```diff
@@ -0,0 +1,21 @@
+class MotdFormatterMarkdownErb
+  attr_reader :content, :title
+
+  def initialize(motd_file)
+    motd_file ||= MotdFile.new unless motd_file
+    @title = motd_file.title
+    @content = OodAppkit.markdown.render(render(motd_file))
+  end
+
+  def render(motd_file)
+    begin
+      ERB.new(motd_file.content).result
+    rescue Exception => e
+      raise "ERB rendering failed with: #{e.message}\n #{motd_file.content}"
+    end
+  end

```

Comment:
OK I know we keep going back and forth here and in writing this comment, I think I convinced myself to go another way. We have 2 paths here.

Either we throw the exception.  We don't rescue here, instead we rescue in the dashboard controller and flash an error message there like this.
![image](https://user-images.githubusercontent.com/4874123/96159267-74ce3180-0ee2-11eb-9d98-bf2326aed7a5.png)

Or we rescue here, and embed the error into the MOTD. So we never throw an exception up the stack from these models. We always return strings, even if rendering fails. The string is just a combination of the error message and the raw content.

![image](https://user-images.githubusercontent.com/4874123/96159443-a47d3980-0ee2-11eb-996c-6e8397c2c20b.png)

I'm wondering what's a better experience for users and operators.  Now I think I'm leaning toward the first option: catching the error in the controller and not rendering the MOTD at all. I think that is a more direct way of telling the user something went wrong whereas the raw ERB content may be confusing and they may not register that as an error. For the operator, I don't think there's any difference really in terms of being able to debug the issue.

Reply:
I like the first option too. Yeah, I'll work on implementing that